### ISSUE TITLE
Reproduce results and some questions about the weight of different loss

### ISSUE THREAD

**Issue by HaoKun-Li** [jiaxiangshang/MGCNet#15]
Hi! Good morning!
I try to use three datasets(CelebA、300W-LP and LS3D-W balanced subset) to train the MGCNet with your code. But I got a much worse result compared with the pre-train model. As shown below:
The results produced by the pre-train model is wonderful.
![image](https://user-images.githubusercontent.com/34161960/94384889-49271980-0176-11eb-90dc-3af2f4863807.png)

But the results from my reproduce model is ugly.
![image](https://user-images.githubusercontent.com/34161960/94384933-60fe9d80-0176-11eb-9806-69f977094632.png)

I followed the training command in  "train_unsupervise.py" without changing any parameter except the batchsize=1. The command is as follow:
![image](https://user-images.githubusercontent.com/34161960/94385055-b5098200-0176-11eb-85e5-de7a3b06e3e7.png)

I think there are some problems during my reproduce:
(1) The step is 20K, which is much smaller than 400K in the paper.
(2) I do not get the Multi-PLE dataset, so the result should be worse than pre-train.
(3) The weight of  epipolar_weight was set to 0.00 in "train_unsupervise.py", I think it is a mistake. In the paper, this weight was set to 0.001. 
(4) I set the batch size to 1 for the limited GPU source. 
(5) Can the code support training with multiple GPUs?

Now I am trying to train again with the same hyper-parameters in the paper.
Thanks for your answers!

**Comment by jiaxiangshang** [comment]
The param in the paper is right, the cmd is just a example.

**Comment by HaoKun-Li** [comment]
I find that the "filling up shuffle buffer" procedure is very slow for a bigger dataset with 16665 training samples. It may take more than one hour to fill up the shuffle buffer before each epoch. Is this phenomenon normal? What can I do to speed up this procedure? 
![image](https://user-images.githubusercontent.com/34161960/94406388-87d3c880-01a4-11eb-8e4b-f1a74918fa0d.png)

**Comment by jiaxiangshang** [comment]
No, it maybe slow, but no one hour. 
This, not a complex operation, and the data is not large.
To speed up, the tensorflow dataset api may help.

**Comment by enjoybo** [comment]
@HaoKun-Li  Hi, can you reproduce the results shown in the paper now?

**Comment by HaoKun-Li** [comment]
> @HaoKun-Li Hi, can you reproduce the results shown in the paper now?

I preprocess the dataset again and try to reproduce the results, but still not yet. But the training code is work, and excellent for learning.  So I think we can reproduce the results in a few days.

**Comment by jiaxiangshang** [comment]
#20 We have a talk about multi-view dataset preprocessing.
This contains some strategy but maybe not a perfect one, hope your work can search on this.
By the way, thanks for your nice words, :)
